tc/sgx/trusted_worker_manager/enclave/work_order_processor.cpp

```diff
@@ -229,11 +228,12 @@ namespace tcf {
                                          d.workorder_data.decrypted_data);
             }
 
-            WorkloadProcessor processor;
             // Convert workload_id from hex string to string
             ByteArray workload_bytes = HexStringToBinary(workload_id);
             std::string workload_type(workload_bytes.begin(), workload_bytes.end());
-            processor.ProcessWorkOrder(
+            WorkloadProcessor *processor = 
+                WorkloadProcessor::CreateWorkloadProcessor(workload_type);
+            processor->ProcessWorkOrder(
```

Comment:
To be on safe side, especially, if CreateWorkloadProcessor() implemenentation changes in the future, add a check
if ( processor == nullptr)
{
     ... through an exception
}

Reply:
ok